### PR TITLE
Added error handling to the dataset api response closing

### PR DESCRIPTION
**Pull request by dandwal**
### What

- Added error handling to the dataset api response closing
- Reordered download service auth token expected parameter position

### How to review

Check this is consistent with the other clients

### Who can review

Anyone except me

### CODE REVIEW

File: dataset/dataset.go [hunk 1]
```diff
@@ -48,6 +49,20 @@ type Client struct {
 	url string
 }
 
+// closeResponseBodyWithContext closes the response body and logs an error containing the context if unsuccessful
+func closeResponseBodyWithContext(ctx context.Context, resp *http.Response) {
+	if err := resp.Body.Close(); err != nil {
+		log.ErrorCtx(ctx, err, log.Data{"Message": "error closing http response body"})
+	}
+}
+// closeResponseBody closes the response body and logs an error if unsuccessful
```
Comment (eldeal) [inline dataset/dataset.go]: comment is on the wrong line/dont need the empty line after it

File: dataset/dataset.go [hunk 2]
```diff
@@ -48,6 +49,20 @@ type Client struct {
 	url string
 }
 
+// closeResponseBodyWithContext closes the response body and logs an error containing the context if unsuccessful
+func closeResponseBodyWithContext(ctx context.Context, resp *http.Response) {
+	if err := resp.Body.Close(); err != nil {
+		log.ErrorCtx(ctx, err, log.Data{"Message": "error closing http response body"})
```
Comment (eldeal) [inline dataset/dataset.go]: I believe logging standards say the keys in log.Data should be lower cased. Not sure whether the aggregator is case sensitive 🤔

File: dataset/dataset.go [hunk 3]
```diff
@@ -48,6 +49,20 @@ type Client struct {
...
+	}
+}
+// closeResponseBody closes the response body and logs an error if unsuccessful
+
+func closeResponseBody(resp *http.Response) {
```
Comment (eldeal) [inline dataset/dataset.go]: do we really need the second function? as it's only used one, I'd suggest only having `func closeResponseBody(ctx, resp)` and just passing nil context in when it's not available? or `context.Background()`

File: dataset/dataset.go (hunk 2 above)
Comment (dandwal) [reply dataset/dataset.go]: Haha, in which case this has been a mistake on all client updates.

**Pull request merged**